### PR TITLE
Socket controller refactoring2

### PR DESCRIPTION
**Pull request by Ai4n**
Please check it!
Only last (this) pull request! 
Reject another ones. ;-)
Thanks,

### CODE REVIEW

File: Server/src/main/Author.java [hunk 1]
```diff
@@ -17,12 +17,12 @@ public Author (int authorId, String name, String surname, Language language) {
 		this.language = language;
 	}
 	
-	public Author (String name, String surname, Language language) {
-		this.name = name;
-		this.surname = surname;
-		this.language = language;
-	}
-	
+//	public Author (String name, String surname, Language language) {
```
Comment (dnlbl) [inline Server/src/main/Author.java]: delete unnecessary lines

File: Server/src/main/ServerController.java [hunk 2]
```diff
@@ -33,8 +27,6 @@ public void handleMessages() {
 			String jsonMessage = socketController.readUtf();
 			SocketExchange request = gson.fromJson(jsonMessage, SocketExchange.class);
 			System.out.println("message: " + request.message);
-			if (request == null)
```
Comment (dnlbl) [inline Server/src/main/ServerController.java:36]: why removed if (request == null) continue;

File: Server/src/main/ServerController.java [hunk 3]
```diff
@@ -72,7 +64,8 @@ public void handleMessages() {
 			case ADD_USER_BOOK:
 				AddBookToUsersBookListRequest addBookToUsersBookListRequest = gson.fromJson(jsonMessage,
 						AddBookToUsersBookListRequest.class);
-				addBookToUsersList(addBookToUsersBookListRequest.getBookId(), addBookToUsersBookListRequest.getUserId());
+				addBookToUsersList(addBookToUsersBookListRequest.getBookId(),
```
Comment (dnlbl) [inline Server/src/main/ServerController.java]: change method's signature from "addBookToUsersList(String, String)" to "addBookToUsersBookListRequest(AddBookToUsersBookListRequest)"

File: Server/src/socketExchange/IsLoginExistResponse.java [hunk 4]
```diff
@@ -1,16 +1,10 @@
...
 
 	public IsLoginExistResponse(Boolean isLoginExist) {
 		super(isLoginExist ? ServerMessage.USER_EXIST : ServerMessage.USER_NOT_EXIST);
 	}
-
-	public String json() {
-		return new Gson().toJson(this);
-	}
 }
+
```
Comment (dnlbl) [inline Server/src/socketExchange/IsLoginExistResponse.java]: add new line at the end of file

File: ServerClient/src/main/AdminController.java [hunk 5]
```diff
@@ -75,8 +75,9 @@ private void deleteUser() {
 		if(number == 0 || number > usersList.size()) {
 			return;
 		}
-		User user = usersList.get(number - 1);
-		DeleteUserRequest deleteUserRequest = new DeleteUserRequest(user.getIdUser());
+		int userId = usersList.get(number - 1).getIdUser();
+		System.out.println(userId);
```
Comment (dnlbl) [inline ServerClient/src/main/AdminController.java]: remove System.out.println

File: ServerClient/src/main/UserController.java [hunk 6]
```diff
@@ -50,9 +50,9 @@ private void showAllUsersBooks() {
 
 	private void addBookToUsersList() {
 		ArrayList<Book> listBooks = getAllLibraryBooks();
-		if (listBooks.equals(null)) {
+		if (listBooks == null) {
 			return;
-		} else if (!(user.equals(null))) {
+		} else if (!(user == null)) {
```
Comment (dnlbl) [inline ServerClient/src/main/UserController.java:57]: "else if" is unnecessary, use "if" instead
change !(user == null) to user != null

File: ServerClient/src/main/UserController.java [hunk 7]
```diff
@@ -61,6 +61,7 @@ private void addBookToUsersList() {
 			}
 			int bookId = listBooks.get(number - 1).getBookId();
 			int userId = user.getIdUser();
+			System.out.println(userId);
```
Comment (dnlbl) [inline ServerClient/src/main/UserController.java]: remove System.out.println

File: ServerClient/src/socketExchange/GetAllUserBooksResponse.java [hunk 8]
```diff
@@ -15,7 +14,5 @@ public GetAllUserBooksResponse (ArrayList<Book> allBooksList) {
 		
 		public ArrayList<Book> getAllBooksList() {
```
Comment (dnlbl) [inline ServerClient/src/socketExchange/GetAllUserBooksResponse.java]: incorrect format;
remove two tabulations, check format of all files

File: Server/src/main/ServerController.java [hunk 9]
```diff
@@ -156,7 +150,7 @@ private void deleteBook(int bookId) {
 	}
 
 	private void deleteUsersBookInList(int bookId, int userId) {
-		bookRepo.deleteUserBook(userId, bookId);
+		userRepo.deleteUserBook(userId, bookId);
```
Comment (dnlbl) [inline Server/src/main/ServerController.java]: why moved method to UserRepo class?

File: ServerClient/src/main/SocketController.java [hunk 10]
```diff
@@ -16,20 +14,6 @@ public SocketController(Socket socket) throws IOException {
...
-//	private String [] convertArrayIntegerToArrayString(int[] arrayInt) {
-//		ArrayList <String> arrayStr = new ArrayList<>();
-//		for(int i = 0; i < arrayInt.length; i++) {
-//			arrayStr.add(String.valueOf(arrayInt[i]));
-//			} 
-//		String[] array = arrayStr.toArray(new String[arrayStr.size()]);
-//		return array;
-//	} 
-//	
 	public void write(String json) {
```
Comment (dnlbl) [inline ServerClient/src/main/SocketController.java]: change method signature from write(String) to write(SocketExchange)

File: ServerClient/src/main/SocketController.java [hunk 11]
```diff
@@ -39,46 +23,6 @@ public void write(String json) {
...
-//	public void writeObject(Object object) {
-//		try {
-//			dataOut.writeObject(object);
-//			dataOut.flush();
-//		} catch (IOException ex) {
-//			ex.printStackTrace();
-//		}
-//	}
-
 	public String readUtf() {
```
Comment (dnlbl) [inline ServerClient/src/main/SocketController.java]: change method signature from readUtf(String) to read(SocketExchange)

**Review by dnlbl — CHANGES_REQUESTED** [review]
needs work

**Review by dnlbl — CHANGES_REQUESTED** [review]
1 issue with "socket" package - models should not be in "controller" package 
correct structure:

socket:
-controller
-model:
--ServerMessage.java
--ServerExchange.java
--socket-exchange (All requests and responses here)

rework in both Server and Client projects

**Comment by Ai4n** [comment]
Please check it

**Comment by Ai4n** [comment]
Fixed

**Review by dnlbl — APPROVED** [review]

**Pull request merged**